### PR TITLE
Add a Comprehend client wrapper so we can mock it and test within our Lambda functions.

### PR DESCRIPTION
**Pull request by pocockn**
### Problem
To make it easier to test our Lambda functions we need to be able to mock the Comprehend client.

### Solution
Add a Comprehend client we can wrap and mock functions within the client.

### Notes

- **Remember** to tag your merge commit after merging 👍
- **Remember** to add the Clubhouse ticket url as a comment ♣️ 🏠

### CODE REVIEW

**Comment by pocockn** [comment]
https://app.clubhouse.io/vidsy/story/7265/hackday-save-free-text-bot-responses

File: comprehend/client_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+						KeyPhrases: keyPhrases,
+					}, nil
+				},
+			}
+
+			client := NewTestClient(mockClient)
+			keyPhraseOutput, err := client.DetectKeyPhrases(&keyPhraseInput)
+
+			assert.Equal(t, keyPhraseOutput.KeyPhrases[0].Text, keyPhrase.Text)
+			if err != nil {
```
Comment (stevenjack) [inline comprehend/client_test.go]: This should be using:

```go
assert.NoError(t, err)
```

File: comprehend/client_test.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			mockClient := &MockComprehendClient{
+				mockDetectKeyPhrases: func(input *comprehendLib.DetectKeyPhrasesInput) (*comprehendLib.DetectKeyPhrasesOutput, error) {
+					return nil, errors.New("Key Phrase Analysis Error")
+				},
+			}
+
+			testClient := NewTestClient(mockClient)
+
+			_, err := testClient.DetectKeyPhrases(&keyPhraseInput)
+			assert.NotNil(t, err)
```
Comment (stevenjack) [inline comprehend/client_test.go]:
```suggestion
			assert.Error(t, err)
```

File: comprehend/client_test.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+					return &comprehendLib.DetectKeyPhrasesOutput{
+						KeyPhrases: keyPhrases,
+					}, nil
+				},
+			}
+
+			client := NewTestClient(mockClient)
+			keyPhraseOutput, err := client.DetectKeyPhrases(&keyPhraseInput)
+
+			assert.Equal(t, keyPhraseOutput.KeyPhrases[0].Text, keyPhrase.Text)
```
Comment (stevenjack) [inline comprehend/client_test.go]: You should be checking that `keyPhraseOutput.KeyPhrases` has more than `0` elements before doing this assertion.

**Pull request merged**